### PR TITLE
Move metrics endpoint listener to use 127.0.0.1 as default

### PR DESCRIPTION
**Pull request by kwilczynski**
/kind feature
/assign kwilczynski

#### What this PR does / why we need it:

Move the metrics endpoint listener to `127.0.0.1` as the default address if no other custom IP address or hostname has been provided. This aims to improve the default security stance in CRI-O when metrics collection is enabled.

While at it, update verbiage of documentation, configuration files and code comments around custom host and port support for metrics.

Related:

- https://github.com/cri-o/cri-o/issues/7446
- https://github.com/cri-o/cri-o/issues/7448
- https://github.com/cri-o/cri-o/pull/7677

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
Move the metrics endpoint listener to use 127.0.0.1 as the new default.
```

### CODE REVIEW

File: test/metrics.bats [hunk 1]
```diff
@@ -11,27 +11,49 @@ function teardown() {
...
+	# Start CRI-O using a custom IP of "127.0.1.1".
+	CONTAINER_ENABLE_METRICS=true CONTAINER_METRICS_HOST=$HOST start_crio
+
+	if ! host_and_port_listens $HOST $PORT; then
+		echo "Metrics endpoint should listen on 127.0.1.1" >&3
+		return 1
+	fi
+
+	curl -sf "http://${HOST}:${PORT}/metrics" | grep crio_operations
 }
```
Comment (saschagrunert) [inline test/metrics.bats:57]: The test failures seems to be legit: 

```
not ok 189 metrics with custom host using 127.0.1.1
# (in test file ./metrics.bats, line 53)
#   `return 1' failed
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/cri-o_cri-o/7714/pull-ci-cri-o-cri-o-main-ci-fedora-integration/1751893021156184064/artifacts/fedora-integration/cri-o-gather/artifacts/testout.txt

Comment (kwilczynski) [reply test/metrics.bats:57]: @saschagrunert, yeah. These have been a hit-and-miss.

I get intermittent failures locally, too. Reason? All run on the same host without isolation.

I think, I will drop some of these tests. Not sure how to fix the concurrent execution issue, aside from setting concurrency to 1 for bats.

Comment (kwilczynski) [reply test/metrics.bats]: Done.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**